code/src/components/AddTodo.js

```diff
@@ -0,0 +1,103 @@
+import React, {useState} from 'react'
+import { useDispatch } from 'react-redux'
+import { v4 as uuidv4 } from 'uuid';
+
+import todos from '../reducers/todos'
+import styled from 'styled-components'
+
+const AddTodo =() => {
+  const [value, setValue] = useState ('')
+  const dispatch = useDispatch()
+
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+
+    const newTodo = {
+      id: uuidv4(), 
+      description: value, 
+      time: Date.now(),
+      isComplete: false,
+    }
+    dispatch(todos.actions.addTodo(newTodo))
+    setValue('')
+  }
+  return(
+    <>
+      <AddTodoText>ADD A NEW TASK AND CLICK ON THE +</AddTodoText>
+      <TodoForm onSubmit={onFormSubmit}>
+        <TodoInput
+          type='text'
+          required
+          value={value}
+          placeholder="Do next . . .✏️"
+          onChange={e => setValue (e.target.value)}
+        />
+        <AddButton type='submit' disabled={
+          value.length < 3 || value.length > 140 }>
+          {" "}+{" "}
+        </AddButton>
+      </TodoForm>
+    </>
+  )
+}
+
+export default AddTodo
+
+const AddTodoText = styled.p`
+color: #ffffff;
```

Comment:
You sometimes use short sometimes long hex color notation. please stay with one convention 😸